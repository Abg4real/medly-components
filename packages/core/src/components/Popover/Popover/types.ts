import { HTMLProps } from '@medly-components/utils';

export interface Props extends HTMLProps<HTMLDivElement> {
    /** Takes full width of the parent component */
    fullWidth?: boolean;
    /** Takes full height of the parent component */
    fullHeight?: boolean;
}
