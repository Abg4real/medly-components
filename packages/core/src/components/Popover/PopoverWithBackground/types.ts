import { HTMLProps } from '@medly-components/utils';
import { Placement } from '../types';

export interface Props extends HTMLProps<HTMLDivElement> {
    /** Takes full width of the parent component */
    fullWidth?: boolean;
    /** Takes full height of the parent component */
    fullHeight?: boolean;
    /** Function to be called on  outer click of the wrapper */
    placement?: Placement;
}
