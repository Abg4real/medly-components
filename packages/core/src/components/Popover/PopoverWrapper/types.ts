import { HTMLProps, Omit } from '@medly-components/utils';

export type Placement =
    | 'top-start'
    | 'top'
    | 'top-end'
    | 'right-start'
    | 'right'
    | 'right-end'
    | 'bottom-end'
    | 'bottom'
    | 'bottom-start'
    | 'left-end'
    | 'left'
    | 'left-start';

export type InteractionType = 'click' | 'hover';

export interface Props extends Omit<HTMLProps<HTMLDivElement>, 'onChange'> {
    /** Pass true to show popover child */
    showPopover?: boolean;
    /** Placement of the popover */
    placement?: Placement;
    /** Function to be called on  outer click of the wrapper */
    onOuterClick?: () => void;
}
