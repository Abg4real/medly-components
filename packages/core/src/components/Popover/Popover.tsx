import { useOuterClickNotifier, WithStyle } from '@medly-components/utils';
import React, { SFC, useRef, useState } from 'react';
import { PopoverStyled, PopoverWrapperStyled } from './Popover.styled';
import { PopoverProps, PopoverWrapperProps } from './types';

export const PopoverWrapper: SFC<PopoverWrapperProps> & WithStyle = props => {
    const wrapperRef = useRef(null);
    const [popoverState, setPopoverState] = useState(false);
    const { onOuterClick, ...restProps } = props;

    const changePopoverState = (state: boolean) => {
        if (props.interactionType === 'click') {
            const length = wrapperRef.current.children.length - 1;
            const popover = wrapperRef.current.children[length] as HTMLElement;
            popover.style.display = state ? 'block' : 'none';
            setPopoverState(state);
        }
    };

    const handleOnClick = () => {
        changePopoverState(!popoverState);
    };

    useOuterClickNotifier(() => {
        changePopoverState(false);
        onOuterClick && onOuterClick();
    }, wrapperRef);

    return <PopoverWrapperStyled {...restProps} ref={wrapperRef} onClick={handleOnClick} data-testid="popover-wrapper" />;
};

PopoverWrapper.defaultProps = {
    interactionType: 'hover',
    placement: 'bottom'
};
PopoverWrapper.Style = PopoverWrapperStyled;

export const Popover: SFC<PopoverProps> & WithStyle = React.forwardRef((props, ref) => <PopoverStyled ref={ref} {...props} />);

Popover.Style = PopoverStyled;
Popover.defaultProps = {
    fullWidth: false,
    fullHeight: false
};
