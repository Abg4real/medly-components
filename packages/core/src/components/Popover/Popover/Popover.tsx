import { WithStyle } from '@medly-components/utils';
import React, { SFC } from 'react';
import { PopoverStyled } from './Popover.styled';
import { Props } from './types';

export const Popover: SFC<Props> & WithStyle = React.memo(
    React.forwardRef((props, ref) => <PopoverStyled ref={ref} {...props} data-testid="medly-popover" />)
);
Popover.displayName = 'Popover';
Popover.Style = PopoverStyled;
Popover.defaultProps = {
    fullWidth: false,
    fullHeight: false
};
