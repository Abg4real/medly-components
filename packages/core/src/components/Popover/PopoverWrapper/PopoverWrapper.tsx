import { useOuterClickNotifier, WithStyle } from '@medly-components/utils';
import React, { SFC, useCallback, useEffect, useRef } from 'react';
import { PopoverWrapperStyled } from './PopoverWrapper.styled';
import { Props } from './types';

export const PopoverWrapper: SFC<Props> & WithStyle = React.memo(props => {
    const wrapperRef = useRef(null),
        { onOuterClick, showPopover, ...restProps } = props,
        id = props.id || 'medly-popover-wrapper';

    const changePopoverState = useCallback(
        (state: boolean) => {
            if (props.interactionType === 'click') {
                const popoverIndex = wrapperRef.current.children.length - 1,
                    popover = wrapperRef.current.children[popoverIndex] as HTMLElement;
                popover.style.display = state ? 'block' : 'none';
            }
        },
        [props.interactionType, wrapperRef.current]
    );

    useEffect(() => {
        changePopoverState(showPopover);
    }, [showPopover]);

    useOuterClickNotifier(() => {
        changePopoverState(false);
        onOuterClick && onOuterClick();
    }, wrapperRef);

    return <PopoverWrapperStyled {...restProps} ref={wrapperRef} id={id} />;
});

PopoverWrapper.displayName = 'PopoverWrapper';
PopoverWrapper.defaultProps = {
    interactionType: 'hover',
    placement: 'bottom'
};
PopoverWrapper.Style = PopoverWrapperStyled;
