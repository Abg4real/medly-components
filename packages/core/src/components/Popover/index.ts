export { default as Popover } from './Popover';
export { default as PopoverWithBackground } from './PopoverWithBackground';
export { default as PopoverWrapper } from './PopoverWrapper';
