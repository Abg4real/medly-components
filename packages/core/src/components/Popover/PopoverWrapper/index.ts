import { PopoverWrapper } from './PopoverWrapper';

export default PopoverWrapper;
