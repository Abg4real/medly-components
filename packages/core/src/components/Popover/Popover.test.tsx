import { TestUtils } from '@medly-components/utils';
import React from 'react';
import Popover from './Popover';
import PopoverWrapper from './PopoverWrapper';
import { Placement, Props } from './PopoverWrapper/types';

const { render, fireEvent, cleanup } = TestUtils;

const renderer = ({ interactionType = 'hover', showPopover = false, placement = 'left', onOuterClick = jest.fn() }: Props) =>
    render(
        <PopoverWrapper {...{ interactionType, showPopover, placement, onOuterClick }}>
            <div>Dummy Div</div>
            <Popover>Dummy popover</Popover>
        </PopoverWrapper>
    );

describe('Popover component', () => {
    afterEach(cleanup);

    describe('with placement', () => {
        afterEach(cleanup);

        test.each([
            'top-start',
            'top',
            'top-end',
            'right-start',
            'right',
            'right-end',
            'bottom-end',
            'bottom',
            'bottom-start',
            'left-end',
            'left',
            'left-start'
        ])('%s, should render properly', (placement: Placement) => {
            const { container, getByTestId } = renderer({ placement });
            fireEvent.mouseOver(getByTestId('medly-popover-wrapper'));
            expect(container).toMatchSnapshot();
        });
    });

    it('should not render popover when interaction type is hover and user clicks on it', () => {
        const { getByTestId } = renderer({ interactionType: 'hover' });
        const wrapper = getByTestId('medly-popover-wrapper'),
            popover = getByTestId('medly-popover');
        fireEvent.click(wrapper);
        expect(popover).not.toBeVisible();
    });

    it('should render popover when interaction type is click and showPopover prop is given', () => {
        const { getByTestId } = renderer({ interactionType: 'click', showPopover: true });
        expect(getByTestId('medly-popover')).toBeVisible();
    });

    it('should not render popover when interaction type is click and showPopover prop is false', () => {
        const { getByTestId } = renderer({ interactionType: 'click', showPopover: false });
        expect(getByTestId('medly-popover')).not.toBeVisible();
    });

    it('should call onOuterClick fn on click on outside', () => {
        const mockOnOuterClick = jest.fn();
        const { getByTestId } = render(
            <div>
                <div data-testid="sibling">Sibling</div>
                <PopoverWrapper interactionType="click" onOuterClick={mockOnOuterClick}>
                    <div>Dummy Div</div>
                    <Popover>Dummy popover</Popover>
                </PopoverWrapper>
            </div>
        );
        fireEvent.click(getByTestId('sibling'));
        expect(mockOnOuterClick).toBeCalled();
    });

    it('should render popover with full width and height if fullWidth and fullHeight are given', () => {
        const { getByTestId } = render(
            <PopoverWrapper interactionType="click" showPopover>
                <div>Dummy Div</div>
                <Popover fullWidth fullHeight>
                    Dummy popover
                </Popover>
            </PopoverWrapper>
        );
        expect(getByTestId('medly-popover')).toHaveStyle(`
            width: calc(100% - 0px);
            height: calc(100% - 0px);
      `);
    });
});
