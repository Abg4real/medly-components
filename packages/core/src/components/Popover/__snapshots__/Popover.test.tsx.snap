// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popover component with placement bottom, should render properly 1`] = `
.c2 {
  position: absolute;
  z-index: 1000;
  display: none;
  -webkit-transition: all 2ms ease-in-out;
  transition: all 2ms ease-in-out;
  border-radius: 4px;
}

.c3 .c1 {
  bottom: calc(100% + 5px);
  left: 0%;
}

.c3:hover .c1 {
  display: block;
}

.c4 .c1 {
  bottom: calc(100% + 5px);
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c4:hover .c1 {
  display: block;
}

.c5 .c1 {
  bottom: calc(100% + 5px);
  right: 0%;
}

.c5:hover .c1 {
  display: block;
}

.c6 .c1 {
  top: 0%;
  left: calc(100% + 5px);
}

.c6:hover .c1 {
  display: block;
}

.c7 .c1 {
  left: calc(100% + 5px);
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c7:hover .c1 {
  display: block;
}

.c8 .c1 {
  bottom: 0px;
  left: calc(100% + 5px);
}

.c8:hover .c1 {
  display: block;
}

.c9 .c1 {
  top: calc(100% + 5px);
  right: 0%;
}

.c9:hover .c1 {
  display: block;
}

.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  position: relative;
  -webkit-flex-direction: column-reverse;
  -ms-flex-direction: column-reverse;
  flex-direction: column-reverse;
}

.c0 .c1 {
  top: calc(100% + 5px);
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c0:hover .c1 {
  display: block;
}

<div>
  <div
    class="c0"
    data-testid="medly-popover-wrapper"
  >
    <div>
      Dummy Div
    </div>
    <div
      class="c1 c2"
      data-testid="medly-popover"
    >
      Dummy popover
    </div>
  </div>
</div>
`;

exports[`Popover component with placement bottom-end, should render properly 1`] = `
.c2 {
  position: absolute;
  z-index: 1000;
  display: none;
  -webkit-transition: all 2ms ease-in-out;
  transition: all 2ms ease-in-out;
  border-radius: 4px;
}

.c3 .c1 {
  bottom: calc(100% + 5px);
  left: 0%;
}

.c3:hover .c1 {
  display: block;
}

.c4 .c1 {
  bottom: calc(100% + 5px);
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c4:hover .c1 {
  display: block;
}

.c5 .c1 {
  bottom: calc(100% + 5px);
  right: 0%;
}

.c5:hover .c1 {
  display: block;
}

.c6 .c1 {
  top: 0%;
  left: calc(100% + 5px);
}

.c6:hover .c1 {
  display: block;
}

.c7 .c1 {
  left: calc(100% + 5px);
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c7:hover .c1 {
  display: block;
}

.c8 .c1 {
  bottom: 0px;
  left: calc(100% + 5px);
}

.c8:hover .c1 {
  display: block;
}

.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  position: relative;
  -webkit-flex-direction: column-reverse;
  -ms-flex-direction: column-reverse;
  flex-direction: column-reverse;
}

.c0 .c1 {
  top: calc(100% + 5px);
  right: 0%;
}

.c0:hover .c1 {
  display: block;
}

<div>
  <div
    class="c0"
    data-testid="medly-popover-wrapper"
  >
    <div>
      Dummy Div
    </div>
    <div
      class="c1 c2"
      data-testid="medly-popover"
    >
      Dummy popover
    </div>
  </div>
</div>
`;

exports[`Popover component with placement bottom-start, should render properly 1`] = `
.c2 {
  position: absolute;
  z-index: 1000;
  display: none;
  -webkit-transition: all 2ms ease-in-out;
  transition: all 2ms ease-in-out;
  border-radius: 4px;
}

.c3 .c1 {
  bottom: calc(100% + 5px);
  left: 0%;
}

.c3:hover .c1 {
  display: block;
}

.c4 .c1 {
  bottom: calc(100% + 5px);
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c4:hover .c1 {
  display: block;
}

.c5 .c1 {
  bottom: calc(100% + 5px);
  right: 0%;
}

.c5:hover .c1 {
  display: block;
}

.c6 .c1 {
  top: 0%;
  left: calc(100% + 5px);
}

.c6:hover .c1 {
  display: block;
}

.c7 .c1 {
  left: calc(100% + 5px);
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c7:hover .c1 {
  display: block;
}

.c8 .c1 {
  bottom: 0px;
  left: calc(100% + 5px);
}

.c8:hover .c1 {
  display: block;
}

.c9 .c1 {
  top: calc(100% + 5px);
  right: 0%;
}

.c9:hover .c1 {
  display: block;
}

.c10 .c1 {
  top: calc(100% + 5px);
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c10:hover .c1 {
  display: block;
}

.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  position: relative;
  -webkit-flex-direction: column-reverse;
  -ms-flex-direction: column-reverse;
  flex-direction: column-reverse;
}

.c0 .c1 {
  top: calc(100% + 5px);
  left: 0%;
}

.c0:hover .c1 {
  display: block;
}

<div>
  <div
    class="c0"
    data-testid="medly-popover-wrapper"
  >
    <div>
      Dummy Div
    </div>
    <div
      class="c1 c2"
      data-testid="medly-popover"
    >
      Dummy popover
    </div>
  </div>
</div>
`;

exports[`Popover component with placement left, should render properly 1`] = `
.c2 {
  position: absolute;
  z-index: 1000;
  display: none;
  -webkit-transition: all 2ms ease-in-out;
  transition: all 2ms ease-in-out;
  border-radius: 4px;
}

.c3 .c1 {
  bottom: calc(100% + 5px);
  left: 0%;
}

.c3:hover .c1 {
  display: block;
}

.c4 .c1 {
  bottom: calc(100% + 5px);
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c4:hover .c1 {
  display: block;
}

.c5 .c1 {
  bottom: calc(100% + 5px);
  right: 0%;
}

.c5:hover .c1 {
  display: block;
}

.c6 .c1 {
  top: 0%;
  left: calc(100% + 5px);
}

.c6:hover .c1 {
  display: block;
}

.c7 .c1 {
  left: calc(100% + 5px);
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c7:hover .c1 {
  display: block;
}

.c8 .c1 {
  bottom: 0px;
  left: calc(100% + 5px);
}

.c8:hover .c1 {
  display: block;
}

.c9 .c1 {
  top: calc(100% + 5px);
  right: 0%;
}

.c9:hover .c1 {
  display: block;
}

.c10 .c1 {
  top: calc(100% + 5px);
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c10:hover .c1 {
  display: block;
}

.c11 .c1 {
  top: calc(100% + 5px);
  left: 0%;
}

.c11:hover .c1 {
  display: block;
}

.c12 .c1 {
  right: calc(100% + 5px);
  bottom: 0%;
}

.c12:hover .c1 {
  display: block;
}

.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  position: relative;
  -webkit-flex-direction: row-reverse;
  -ms-flex-direction: row-reverse;
  flex-direction: row-reverse;
}

.c0 .c1 {
  right: calc(100% + 5px);
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c0:hover .c1 {
  display: block;
}

<div>
  <div
    class="c0"
    data-testid="medly-popover-wrapper"
  >
    <div>
      Dummy Div
    </div>
    <div
      class="c1 c2"
      data-testid="medly-popover"
    >
      Dummy popover
    </div>
  </div>
</div>
`;

exports[`Popover component with placement left-end, should render properly 1`] = `
.c2 {
  position: absolute;
  z-index: 1000;
  display: none;
  -webkit-transition: all 2ms ease-in-out;
  transition: all 2ms ease-in-out;
  border-radius: 4px;
}

.c3 .c1 {
  bottom: calc(100% + 5px);
  left: 0%;
}

.c3:hover .c1 {
  display: block;
}

.c4 .c1 {
  bottom: calc(100% + 5px);
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c4:hover .c1 {
  display: block;
}

.c5 .c1 {
  bottom: calc(100% + 5px);
  right: 0%;
}

.c5:hover .c1 {
  display: block;
}

.c6 .c1 {
  top: 0%;
  left: calc(100% + 5px);
}

.c6:hover .c1 {
  display: block;
}

.c7 .c1 {
  left: calc(100% + 5px);
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c7:hover .c1 {
  display: block;
}

.c8 .c1 {
  bottom: 0px;
  left: calc(100% + 5px);
}

.c8:hover .c1 {
  display: block;
}

.c9 .c1 {
  top: calc(100% + 5px);
  right: 0%;
}

.c9:hover .c1 {
  display: block;
}

.c10 .c1 {
  top: calc(100% + 5px);
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c10:hover .c1 {
  display: block;
}

.c11 .c1 {
  top: calc(100% + 5px);
  left: 0%;
}

.c11:hover .c1 {
  display: block;
}

.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  position: relative;
  -webkit-flex-direction: row-reverse;
  -ms-flex-direction: row-reverse;
  flex-direction: row-reverse;
}

.c0 .c1 {
  right: calc(100% + 5px);
  bottom: 0%;
}

.c0:hover .c1 {
  display: block;
}

<div>
  <div
    class="c0"
    data-testid="medly-popover-wrapper"
  >
    <div>
      Dummy Div
    </div>
    <div
      class="c1 c2"
      data-testid="medly-popover"
    >
      Dummy popover
    </div>
  </div>
</div>
`;

exports[`Popover component with placement left-start, should render properly 1`] = `
.c2 {
  position: absolute;
  z-index: 1000;
  display: none;
  -webkit-transition: all 2ms ease-in-out;
  transition: all 2ms ease-in-out;
  border-radius: 4px;
}

.c3 .c1 {
  bottom: calc(100% + 5px);
  left: 0%;
}

.c3:hover .c1 {
  display: block;
}

.c4 .c1 {
  bottom: calc(100% + 5px);
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c4:hover .c1 {
  display: block;
}

.c5 .c1 {
  bottom: calc(100% + 5px);
  right: 0%;
}

.c5:hover .c1 {
  display: block;
}

.c6 .c1 {
  top: 0%;
  left: calc(100% + 5px);
}

.c6:hover .c1 {
  display: block;
}

.c7 .c1 {
  left: calc(100% + 5px);
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c7:hover .c1 {
  display: block;
}

.c8 .c1 {
  bottom: 0px;
  left: calc(100% + 5px);
}

.c8:hover .c1 {
  display: block;
}

.c9 .c1 {
  top: calc(100% + 5px);
  right: 0%;
}

.c9:hover .c1 {
  display: block;
}

.c10 .c1 {
  top: calc(100% + 5px);
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c10:hover .c1 {
  display: block;
}

.c11 .c1 {
  top: calc(100% + 5px);
  left: 0%;
}

.c11:hover .c1 {
  display: block;
}

.c12 .c1 {
  right: calc(100% + 5px);
  bottom: 0%;
}

.c12:hover .c1 {
  display: block;
}

.c13 .c1 {
  right: calc(100% + 5px);
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c13:hover .c1 {
  display: block;
}

.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  position: relative;
  -webkit-flex-direction: row-reverse;
  -ms-flex-direction: row-reverse;
  flex-direction: row-reverse;
}

.c0 .c1 {
  right: calc(100% + 5px);
  top: 0%;
}

.c0:hover .c1 {
  display: block;
}

<div>
  <div
    class="c0"
    data-testid="medly-popover-wrapper"
  >
    <div>
      Dummy Div
    </div>
    <div
      class="c1 c2"
      data-testid="medly-popover"
    >
      Dummy popover
    </div>
  </div>
</div>
`;

exports[`Popover component with placement right, should render properly 1`] = `
.c2 {
  position: absolute;
  z-index: 1000;
  display: none;
  -webkit-transition: all 2ms ease-in-out;
  transition: all 2ms ease-in-out;
  border-radius: 4px;
}

.c3 .c1 {
  bottom: calc(100% + 5px);
  left: 0%;
}

.c3:hover .c1 {
  display: block;
}

.c4 .c1 {
  bottom: calc(100% + 5px);
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c4:hover .c1 {
  display: block;
}

.c5 .c1 {
  bottom: calc(100% + 5px);
  right: 0%;
}

.c5:hover .c1 {
  display: block;
}

.c6 .c1 {
  top: 0%;
  left: calc(100% + 5px);
}

.c6:hover .c1 {
  display: block;
}

.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  position: relative;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c0 .c1 {
  left: calc(100% + 5px);
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c0:hover .c1 {
  display: block;
}

<div>
  <div
    class="c0"
    data-testid="medly-popover-wrapper"
  >
    <div>
      Dummy Div
    </div>
    <div
      class="c1 c2"
      data-testid="medly-popover"
    >
      Dummy popover
    </div>
  </div>
</div>
`;

exports[`Popover component with placement right-end, should render properly 1`] = `
.c2 {
  position: absolute;
  z-index: 1000;
  display: none;
  -webkit-transition: all 2ms ease-in-out;
  transition: all 2ms ease-in-out;
  border-radius: 4px;
}

.c3 .c1 {
  bottom: calc(100% + 5px);
  left: 0%;
}

.c3:hover .c1 {
  display: block;
}

.c4 .c1 {
  bottom: calc(100% + 5px);
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c4:hover .c1 {
  display: block;
}

.c5 .c1 {
  bottom: calc(100% + 5px);
  right: 0%;
}

.c5:hover .c1 {
  display: block;
}

.c6 .c1 {
  top: 0%;
  left: calc(100% + 5px);
}

.c6:hover .c1 {
  display: block;
}

.c7 .c1 {
  left: calc(100% + 5px);
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c7:hover .c1 {
  display: block;
}

.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  position: relative;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c0 .c1 {
  bottom: 0px;
  left: calc(100% + 5px);
}

.c0:hover .c1 {
  display: block;
}

<div>
  <div
    class="c0"
    data-testid="medly-popover-wrapper"
  >
    <div>
      Dummy Div
    </div>
    <div
      class="c1 c2"
      data-testid="medly-popover"
    >
      Dummy popover
    </div>
  </div>
</div>
`;

exports[`Popover component with placement right-start, should render properly 1`] = `
.c2 {
  position: absolute;
  z-index: 1000;
  display: none;
  -webkit-transition: all 2ms ease-in-out;
  transition: all 2ms ease-in-out;
  border-radius: 4px;
}

.c3 .c1 {
  bottom: calc(100% + 5px);
  left: 0%;
}

.c3:hover .c1 {
  display: block;
}

.c4 .c1 {
  bottom: calc(100% + 5px);
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c4:hover .c1 {
  display: block;
}

.c5 .c1 {
  bottom: calc(100% + 5px);
  right: 0%;
}

.c5:hover .c1 {
  display: block;
}

.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  position: relative;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c0 .c1 {
  top: 0%;
  left: calc(100% + 5px);
}

.c0:hover .c1 {
  display: block;
}

<div>
  <div
    class="c0"
    data-testid="medly-popover-wrapper"
  >
    <div>
      Dummy Div
    </div>
    <div
      class="c1 c2"
      data-testid="medly-popover"
    >
      Dummy popover
    </div>
  </div>
</div>
`;

exports[`Popover component with placement top, should render properly 1`] = `
.c2 {
  position: absolute;
  z-index: 1000;
  display: none;
  -webkit-transition: all 2ms ease-in-out;
  transition: all 2ms ease-in-out;
  border-radius: 4px;
}

.c3 .c1 {
  bottom: calc(100% + 5px);
  left: 0%;
}

.c3:hover .c1 {
  display: block;
}

.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  position: relative;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c0 .c1 {
  bottom: calc(100% + 5px);
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c0:hover .c1 {
  display: block;
}

<div>
  <div
    class="c0"
    data-testid="medly-popover-wrapper"
  >
    <div>
      Dummy Div
    </div>
    <div
      class="c1 c2"
      data-testid="medly-popover"
    >
      Dummy popover
    </div>
  </div>
</div>
`;

exports[`Popover component with placement top-end, should render properly 1`] = `
.c2 {
  position: absolute;
  z-index: 1000;
  display: none;
  -webkit-transition: all 2ms ease-in-out;
  transition: all 2ms ease-in-out;
  border-radius: 4px;
}

.c3 .c1 {
  bottom: calc(100% + 5px);
  left: 0%;
}

.c3:hover .c1 {
  display: block;
}

.c4 .c1 {
  bottom: calc(100% + 5px);
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c4:hover .c1 {
  display: block;
}

.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  position: relative;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c0 .c1 {
  bottom: calc(100% + 5px);
  right: 0%;
}

.c0:hover .c1 {
  display: block;
}

<div>
  <div
    class="c0"
    data-testid="medly-popover-wrapper"
  >
    <div>
      Dummy Div
    </div>
    <div
      class="c1 c2"
      data-testid="medly-popover"
    >
      Dummy popover
    </div>
  </div>
</div>
`;

exports[`Popover component with placement top-start, should render properly 1`] = `
.c2 {
  position: absolute;
  z-index: 1000;
  display: none;
  -webkit-transition: all 2ms ease-in-out;
  transition: all 2ms ease-in-out;
  border-radius: 4px;
}

.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  position: relative;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c0 .c1 {
  bottom: calc(100% + 5px);
  left: 0%;
}

.c0:hover .c1 {
  display: block;
}

<div>
  <div
    class="c0"
    data-testid="medly-popover-wrapper"
  >
    <div>
      Dummy Div
    </div>
    <div
      class="c1 c2"
      data-testid="medly-popover"
    >
      Dummy popover
    </div>
  </div>
</div>
`;
